ui/src/AzureUserInput.ts

```diff
@@ -18,23 +18,27 @@ export class AzureUserInput implements IAzureUserInput {
         this._persistence = persistence;
     }
 
-    public async showQuickPick<T extends QuickPickItem>(items: T[] | Thenable<T[]>, options: QuickPickOptions): Promise<T> {
+    public async showQuickPick<T extends QuickPickItem>(items: T[] | Thenable<T[]>, options: QuickPickOptions): Promise<T>;
+    public async showQuickPick<T extends QuickPickItem>(items: T[] | Thenable<T[]>, options: QuickPickOptions & { canPickMany: true }): Promise<T[]> {
         if (options.ignoreFocusOut === undefined) {
             options.ignoreFocusOut = true;
         }
 
         let persistenceKey: string | undefined;
         const unhashedKey: string | undefined = (<IAzureQuickPickOptions>options).id || options.placeHolder;
-        if (unhashedKey) {
+        if (unhashedKey && !options.canPickMany) {
             persistenceKey = `showQuickPick.${randomUtils.getPseudononymousStringHash(unhashedKey)}`;
         }
 
-        const result: T | undefined = await vscode.window.showQuickPick(this.getOrderedItems(items, persistenceKey, (<IAzureQuickPickOptions>options).suppressPersistence), options);
-        if (result === undefined) {
+        const result: T[] | undefined = await vscode.window.showQuickPick(this.getOrderedItems(items, persistenceKey, (<IAzureQuickPickOptions>options).suppressPersistence), options);
+        if (result === undefined || result.length === 0) {
             throw new UserCancelledError();
         }
+        if (Array.isArray(result)) {
+            return result;
+        }
 
-        if (persistenceKey && !(<IAzureQuickPickItem><{}>result).suppressPersistence) {
+        if (persistenceKey && !(<IAzureQuickPickItem><{}>result).suppressPersistence && !Array.isArray(result)) {
```

Comment:
Not needed - you've already returned in this case.

Reply:
I put that in so that the compiler knew that result being used in the next line was not an array. It felt like code that informs the compiler rather than function.

Reply:
I agree with @StephenWeatherford feels like this shouldn't matter even for the compiler